integrations/amplitude/lib/index.js

```diff
@@ -126,6 +125,8 @@ Amplitude.prototype.loaded = function() {
 Amplitude.prototype.page = function(page) {
   this.setDeviceIdFromAnonymousId(page);
 
+  if (this.options.trackReferrer) this.sendReferrer();
```

Comment:
Note this occurs right _after_ `setDeviceIdFromAnonymousId` so that the correct DeviceId is used.